pkg/util/helper/cache_test.go

```diff
@@ -0,0 +1,304 @@
+					return m
+				}(),
+				manager: func(stopCh <-chan struct{}) genericmanager.MultiClusterInformerManager {
+					return genericmanager.NewMultiClusterInformerManager(stopCh)
+				},
+				fedKey: keys.FederatedKey{Cluster: "cluster", ClusterWideKey: keys.ClusterWideKey{
+					Version: "v1", Kind: "Pod", Namespace: "default", Name: "pod",
+				}},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name: "get from client, and not found",
+			args: args{
+				restMapper: func() meta.RESTMapper {
+					m := meta.NewDefaultRESTMapper([]schema.GroupVersion{corev1.SchemeGroupVersion})
+					m.Add(schema.GroupVersionKind{Version: "v1", Kind: "Pod"}, meta.RESTScopeNamespace)
+					return m
+				}(),
+				manager: func(stopCh <-chan struct{}) genericmanager.MultiClusterInformerManager {
+					m := genericmanager.NewMultiClusterInformerManager(stopCh)
+					m.ForCluster("cluster", fake.NewSimpleDynamicClient(scheme.Scheme), 0)
+					return m
+				},
+				fedKey: keys.FederatedKey{Cluster: "cluster", ClusterWideKey: keys.ClusterWideKey{
+					Version: "v1", Kind: "Pod", Namespace: "default", Name: "pod",
+				}},
+			},
+			want:    nil,
+			wantErr: true,
+		},
+		{
+			name: "get from client, and get success",
+			args: args{
+				restMapper: func() meta.RESTMapper {
+					m := meta.NewDefaultRESTMapper([]schema.GroupVersion{corev1.SchemeGroupVersion})
+					m.Add(schema.GroupVersionKind{Version: "v1", Kind: "Pod"}, meta.RESTScopeNamespace)
+					return m
+				}(),
+				manager: func(stopCh <-chan struct{}) genericmanager.MultiClusterInformerManager {
+					m := genericmanager.NewMultiClusterInformerManager(stopCh)
+					m.ForCluster("cluster", fake.NewSimpleDynamicClient(scheme.Scheme,
+						&corev1.Pod{ObjectMeta: metav1.ObjectMeta{Name: "pod", Namespace: "default"}},
+					), 0)
+					return m
+				},
+				fedKey: keys.FederatedKey{Cluster: "cluster", ClusterWideKey: keys.ClusterWideKey{
+					Version: "v1", Kind: "Pod", Namespace: "default", Name: "pod",
+				}},
+			},
+			want: &unstructured.Unstructured{Object: map[string]interface{}{
+				"apiVersion": "v1", "kind": "Pod",
+				"metadata": map[string]interface{}{"name": "pod", "namespace": "default", "creationTimestamp": nil},
+				"spec":     map[string]interface{}{"containers": nil},
+				"status":   map[string]interface{}{},
+			}},
+			wantErr: false,
+		},
+		{
+			name: "get from cache, and not found",
+			args: args{
+				restMapper: func() meta.RESTMapper {
+					m := meta.NewDefaultRESTMapper([]schema.GroupVersion{corev1.SchemeGroupVersion})
+					m.Add(corev1.SchemeGroupVersion.WithKind("Pod"), meta.RESTScopeNamespace)
+					return m
+				}(),
+				manager: func(stopCh <-chan struct{}) genericmanager.MultiClusterInformerManager {
+					m := genericmanager.NewMultiClusterInformerManager(stopCh)
+					m.ForCluster("cluster", fake.NewSimpleDynamicClient(scheme.Scheme), 0).
+						Lister(corev1.SchemeGroupVersion.WithResource("pods"))
```

Comment:
Just a question: why need to call `Lister`, I try with removing it and it works ok.